artman/tasks/python_grpc_tasks.py

```diff
@@ -94,35 +94,41 @@ def _extract_base_dirs(self, proto_file):
             return ''
 
     def _transform(self, pkg, sep, common_protos):
-        """Add 'proto' package after 'google' or 'google.cloud'
+        """Transform to the appropriate proto package layout.
 
         Works with arbitrary separator (e.g., '/' for import statements,
         '.' for proto package statements, os.path.sep for filenames)
         """
-        # Skip common protos
-        pkg_list = pkg.split(sep)
-
-        dotted = '.'.join(pkg_list)
+        if sep != '.' and pkg.endswith('.proto'):
+            dotted = pkg[:-6].replace(sep, '.')
+            suffix = '.proto'
+        else:
+            dotted = pkg.replace(sep, '.')
+            suffix = ''
+
+        # Sanity check: Do not transform common protos.
         for common_pkg in common_protos:
             if dotted.startswith(common_pkg):
-                return sep.join(pkg_list)
+                return pkg
+
+        # Transform into the ideal proto path.
+        # What essentially should happen here is that "{api}.{vN}" should
+        # change to "{api}_{vN}.proto".
+        dotted = re.sub(r'\.v([\da-z_]*)([\d]+)\b', r'_v\1\2.proto', dotted)
```

Comment:
We have internal customers who use `v1alpha`, which I think this doesn't match (see: https://github.com/googleapis/toolkit/issues/1335). Ideally this should mirror the version matcher used in api-compiler: https://github.com/googleapis/api-compiler/blob/master/src/main/java/com/google/api/tools/framework/aspects/versioning/model/ApiVersionUtil.java#L40

Also, you really only want to replace the rightmost occurrence of this match, but maybe we don't need to worry about that given how rare it should be to have any other package match the version regex.

Reply:
It would be good to note that the apiVersionUtil regex has not been updated to use the point release syntax.

Reply:
Good catch! (I agree that we do not need to worry about the right-most issue.)

Reply:
Updated in 421db76.